3-networks/envs/shared/hierarchical_firewall.tf

```diff
@@ -0,0 +1,102 @@
+/**
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+module "hierarchical_firewall_policy" {
+  source = "../../modules/hierarchical_firewall_policy/"
+  parent = data.google_active_folder.common.name
+  name   = "common-firewall-rules"
+  associations = [
+    data.google_active_folder.common.name,
+    data.google_active_folder.bootstrap.name,
+    data.google_active_folder.development.name,
+    data.google_active_folder.production.name,
+    data.google_active_folder.non-production.name,
+  ]
+  rules = {
+    delegate-rfc1918-ingress = {
+      description = "Delegate RFC1918 ingress"
+      direction   = "INGRESS"
+      action      = "goto_next"
+      priority    = 500
+      ranges = [
+        "192.168.0.0/16",
+        "10.0.0.0/8",
+        "172.16.0.0/12"
+      ]
+      ports                   = { "all" = [] }
+      target_service_accounts = null
+      target_resources        = null
+      logging                 = false
+    }
+    delegate-rfc1918-egress = {
+      description = "Delegate RFC1918 egress"
+      direction   = "EGRESS"
+      action      = "goto_next"
+      priority    = 510
+      ranges = [
+        "192.168.0.0/16",
+        "10.0.0.0/8",
+        "172.16.0.0/12"
+      ]
+      ports                   = { "all" = [] }
+      target_service_accounts = null
+      target_resources        = null
+      logging                 = false
+    }
+    allow-iap-ssh-rdp = {
+      description             = "Always allow SSH and RDP from IAP"
+      direction               = "INGRESS"
+      action                  = "allow"
+      priority                = 5000
+      ranges                  = ["35.235.240.0/20"]
+      ports                   = { "all" = [] }
+      target_service_accounts = null
+      target_resources        = null
+      logging                 = var.firewall_policies_enable_logging
+    }
+    allow-windows-activation = {
```

Comment:
without the target tags, how would users make use of these rules? Is the default that this will apply to all instances if no target resource or service accounts are supplied?

Reply:
I believe it's inherited and will be applied to all unless overridden with a regular FW rule. Some patterns here https://cloud.google.com/vpc/docs/firewall-policies-examples#example_4_configure_organization-wide_and_folder-specific_rules

Reply:
These rules (`allow-iap-ssh-rdp`, `allow-windows-activation`, `allow-google-hbs-and-hcs`) would always be applied. The only one that can be overridden is `delegate-rfc1918-ingress`. 

Check out section 7.6.1 of the draft document.